test-app/tests/integration/components/input-field-test.gts

```diff
@@ -28,19 +27,33 @@ module('Integration | Component | InputField', function (hooks) {
     assert.dom(input).hasNoClass('shadow-error-outline');
   });
 
-  test('it encourages @label via assert', async function (assert) {
+  test('it throws an assertion error if no `@label` or `:label` is provided', async function (assert) {
     assert.expect(1);
 
     setupOnerror((e: Error) => {
       assert.strictEqual(
         e.message,
-        'Assertion Failed: input field requires a label',
+        'Assertion Failed: You need either :label or @label',
+        'Expected an error message if @label is not provided'
+      );
+    });
+    await render(<template><InputField type="text" /></template>);
+  });
+
+  test('it throws an assertion error if both `@label` and `:label` are provided', async function (assert) {
+    assert.expect(1);
+
+    setupOnerror((e: Error) => {
+      assert.strictEqual(
+        e.message,
+        'Assertion Failed: You can have :label or @label, but not both',
         'Expected an error message if @label is not provided'
```

Comment:
nit: It looks like this message is now incorrect.  Looks like the others do this to be a bit generic:

```suggestion
        'Expected assertion error message'
```